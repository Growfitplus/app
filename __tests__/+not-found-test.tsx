import NotFoundScreen from '@/app/+not-found';
import { render } from '@testing-library/react-native';

describe('Not Found', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<NotFoundScreen />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { getByText } = render(<NotFoundScreen />);

    // eslint-disable-next-line quotes
    expect(getByText("This screen doesn't exist.")).toBeTruthy();
  });

  it('contains a link to the home screen', () => {
    const { getByText } = render(<NotFoundScreen />);

    expect(getByText('Go to home screen!')).toBeTruthy();
  });
});
