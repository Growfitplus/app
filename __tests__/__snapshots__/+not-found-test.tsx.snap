// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text>
    This screen doesn't exist.
  </Text>
  <Text
    href="/"
    onPress={[Function]}
    role="link"
    style={
      {
        "marginTop": 15,
        "paddingVertical": 15,
      }
    }
  >
    <Text>
      Go to home screen!
    </Text>
  </Text>
</View>
`;
